src/lib/services/contentful/fetch.test.ts

```diff
@@ -0,0 +1,57 @@
+import { describe, expect, it, vi, type Mock } from "vitest";
+import * as envVars from "$env/static/private";
+import contentfulFetch from "./fetch";
+
+vi.mock("$env/static/private");
+type EnvVarName = "CONTENTFUL_SPACE_ID" | "CONTENTFUL_DELIVERY_API_TOKEN";
+type EnvVars = Record<EnvVarName, string>;
+const setEnvVars = (vars: EnvVars) => {
+  let key: keyof EnvVars;
+  for (key in vars) {
+    (envVars as EnvVars)[key] = vars[key];
+  }
+};
+
+const query = "{ some { graphql { query } } }";
+
+describe("Contentful Fetch", () => {
+  beforeEach(() => {
+    // Re-initialize fetch mock every time (since we're clearing it every time).
+    global.fetch = vi.fn(async () => Promise.resolve(true)) as Mock;
+  });
+
+  afterEach(() => {
+    // Clear mock history and reset implementations to empty functions.
+    vi.resetAllMocks();
```

Comment:
Nit: if you use `vi.restoreAllMocks()` here instead of `vi.resetAllMocks()`, you can initialize `fetch` once instead of in a `beforeEach`. Either way works tho.

Reply:
Ah you're right, this was leftover from a previous implementation where I was mocking the env vars a different way. I think we can actually just mock fetch once at the top and use `vi.clearAllMocks()` as long as we're going to call `setEnvVars` at the top of each test case.

I'll update this and make a note of it.